.github/workflows/precommit.yaml

```diff
@@ -0,0 +1,47 @@
+name: "Precommit tests"
```

Comment:
Do you think we'll be able to use the same file for Windows testing?  If not, I would call this "Linux Precommit Tests" and change the filename too.